embassy-executor-macros/src/macros/main.rs

```diff
@@ -112,17 +125,44 @@ pub fn run(args: TokenStream, item: TokenStream, arch: &Arch) -> TokenStream {
         error(&mut errors, &f.sig, "main function must have 1 argument: the spawner.");
     }
 
-    let entry = match args.entry.as_deref().or(arch.default_entry) {
-        None => TokenStream::new(),
-        Some(x) => match TokenStream::from_str(x) {
+    let entry = match (&args.entry.as_deref(), &arch.default_entry) {
+        (None, Entry::NotRequired) => TokenStream::new(),
+        (Some(x), _) | (None, Entry::Some(x)) if *x == "" => TokenStream::new(),
```

Comment:
what's the point of allowing `""`?

Reply:
A function called directly by assembly or some other initialization code that isn't a full HAL with proc macros

Reply:
yes, but can't the user simply not specify `entry` in that case?

Reply:
The reason I made specifying it a requirement is to show an error message that is easier to understand than a link error from a missing entrypoint, which is what would happen if entry wasn't supplied or defaulted by an arch feature.